### ISSUE TITLE
[Improvement] record the whole fields while the transaction failing

### ISSUE THREAD

**Issue by nickhsine** [twreporter/go-api#256]
In Narwhale, we cannot see the whole transaction information if the transaction has `fail` status.
Current code logic is in [these lines for periodic donations](https://github.com/twreporter/go-api/blob/master/controllers/donation.go#L494-L498), and [these lines for one-time donations](https://github.com/twreporter/go-api/blob/master/controllers/donation.go#L575-L578)
We only record `tapPayResp.Status ` and  `tapPayResp.Msg`, lacks of other important information, such as `rec_trade_id`, `bank_result_code `, `bank_result_msg ` ...etc.

**Comment by babygoat** [comment]
Any use case for these fields?  It seemed to me that the useful information inferred from `msg`.

**Comment by nickhsine** [comment]
`bank_result_msg` and `bank_result_code` are pretty essential for editors and me for debugging.
Otherwise, I need to go to TapPay dashboard to check the whole information.
Since we have the chance to record them, why don't?

**Comment by nickhsine** [comment]
Take the prime donation(id: 229) for example.

We only record 
```
tappay_api_status  10003
TapPay錯誤訊息  Card error, please ask for issuer
order_number  twreporter-156050575109308397100
--

```

However, in TapPay dashboard,
we can get these information
```
使用者資料
前六後四卡號 ******-**** (* for protecting privacy)
Card Identifier ******************************* 
交易識別碼(recTradeId) **************
訂單編號  **************
銀行錯誤訊息: Bank Error
銀行錯誤代碼: L005
```

When we get L005, we can google this error code(L005) and get `L005 扣款帳號錯誤` from https://portal.tbb.com.tw/tbbportal/fstop/html/NNB_QA.html

Even though `L005 扣款帳號錯誤` maybe not the right answer, but it at least provides some hints.

**Comment by babygoat** [comment]
Go through the backend api [error code](https://docs.tappaysdk.com/tutorial/zh/error.html#api-error-code), we will find that it covers most of the error cases we can infer from the transaction.

What's more, `msg` field as well as the `status` serves for the error dto(data transfer object) from bank transaction to our backend  so these cases are what we should deal with for error handling even for the marketing to do the customer service. This was my original design intention.

If the customer service and we should find these detailed information useful, I can append these fields as improvement.

**Comment by nickhsine** [comment]
The donor sent a email to sylvie, asking why she can't donate by credit cards?

Sylvie just asked me the same question.
However, I couldn't tell any possible reasons except from Msg and Status.

We will need this improvement for further debugging and customer service.

By the way, after your explanation, I clearly know your design intention.

I will change the issue title for better understanding.

**Comment by nickhsine** [comment]
@babygoat 
Is this issue fixed? 
If so, please help to close it.

**Comment by babygoat** [comment]
Resolved with #257 and twreporter/pangolin#97.
